lib/storage/file.js

```diff
@@ -104,8 +104,44 @@ var STORAGE_UPLOAD_BASE_URL = 'https://www.googleapis.com/upload/storage/v1/b';
  * var file = myBucket.file('my-file');
  */
 function File(bucket, name, options) {
+  options = options || {};
+
+  this.bucket = bucket;
+  this.storage = bucket.parent;
+
+  Object.defineProperty(this, 'name', {
+    enumerable: true,
+    value: name
+  });
+
+  var requestQueryObject = {};
+
+  if (is.number(options.generation)) {
```

Comment:
This works? I was looking at [`is.number`](https://github.com/enricomarino/is/blob/master/index.js#L412) and I thought `NaN` would be considered a number by that logic.